ts/twitter/src/app/login/_components/account-modal.tsx

```diff
@@ -1,23 +1,103 @@
 "use client";
 
-import { Button } from "@chakra-ui/react";
+import theme from "@/lib/theme";
+import { Button, Divider, Input, Text, VStack } from "@chakra-ui/react";
+import { useState } from "react";
 import type React from "react";
+import { FaApple, FaGoogle } from "react-icons/fa";
 
 interface AccountModalProps {
-  onNext: () => void;
+  onNext: (username: string) => void;
 }
 
 export const AccountModal: React.FC<AccountModalProps> = ({ onNext }) => {
+  const [username, setUsername] = useState("");
+
+  const handleNext = (e: React.FormEvent) => {
+    e.preventDefault();
+    if (username.trim() !== "") {
+      onNext(username);
+    } else {
+      alert("Please enter a username.");
+    }
+  };
+
   return (
-    <Button
-      onClick={onNext}
-      fontSize="lg"
-      borderRadius="full"
-      bg="white"
-      color="black"
-      fontWeight="bold"
-    >
-      Next
-    </Button>
+    <VStack spacing={4} as="form" onSubmit={handleNext}>
```

Comment:
```suggestion
    <VStack as="form" spacing={4} onSubmit={handleNext}>
```
nit: It's better to place the "as" property first for better clarity.